src/turbomind/layers/DynamicDecodeLayer.cc

```diff
@@ -35,7 +37,9 @@ DynamicDecodeLayer::DynamicDecodeLayer(DataType              dtype,
     TM_CHECK(dtype == kFloat32);
     BaseDynamicDecodeLayer::BaseParam param{max_batch_size, vocab_size, vocab_size_padded, stream, device_prop};
     layers_.emplace_back(new LogitsProcessorLayer<float>{param});
+    layers_.emplace_back(new GuidedDecodeMaskLayer<float>{param});
     layers_.emplace_back(new SamplingLayer<float>{param});
+    layers_.emplace_back(new GuidedDecodeUpdateLayer<float>{param});
     layers_.emplace_back(new StopCriteriaLayer<float>{param});
```

Comment:
The sampling-related classes are declared as templates, but the template parameter `T` does not appear to be utilized in any of the following:
- Member variable
- Member function
- Base class
how about remove the templates? @lzhangzz @irexyc  any comments?